models/index.js

```diff
@@ -1,67 +1,91 @@
-    
-    return db.query(queryStr).then(({ rows }) => {
-        return rows;
-      });
-}
+  const queryStr = `SELECT articles.author, title, articles.article_id, topic, articles.created_at, article_img_url, COUNT(comments.article_id) AS comment_count, SUM(comments.votes) AS votes FROM articles JOIN comments ON articles.article_id=comments.article_id GROUP BY articles.article_id ORDER BY articles.created_at;`;
+
+  return db.query(queryStr).then(({ rows }) => {
+    return rows;
+  });
+};
 
 exports.selectCommentsById = (article_id) => {
-    const regex=/\D/
-    const isNonDigit=regex.test(article_id)
-    if (article_id<=0 || isNonDigit) {
-        return Promise.reject({ status: 400, msg: "Bad request" })
-    }
-    
-    const queryStr = `SELECT * FROM comments WHERE comments.article_id = $1 ORDER BY comments.created_at DESC;`
-    
-    return db.query(queryStr,[article_id]).then(({ rows }) => {
-        return rows;
-      });
-}
+  const regex = /\D/;
+  const isNonDigit = regex.test(article_id);
+  if (article_id <= 0 || isNonDigit) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+
+  const queryStr = `SELECT * FROM comments WHERE comments.article_id = $1 ORDER BY comments.created_at DESC;`;
+
+  return db.query(queryStr, [article_id]).then(({ rows }) => {
+    return rows;
+  });
+};
 
 exports.checkArticleExists = (article_id) => {
-    const queryStr = `SELECT * FROM articles WHERE article_id = $1 ;`
-    return db.query(queryStr,[article_id]).then(({ rows }) => {
-        if(rows.length===0){
-            return Promise.reject({ status: 404, msg: "Article not found" });
-        }
-      });
-}
+  const queryStr = `SELECT * FROM articles WHERE article_id = $1 ;`;
+  return db.query(queryStr, [article_id]).then(({ rows }) => {
+    if (rows.length === 0) {
+      return Promise.reject({ status: 404, msg: "Article not found" });
+    }
+  });
+};
 
 exports.checkArticlesExists = () => {
-    const queryStr = format(`SELECT * FROM articles;`)
-    return db.query(queryStr).then(({ rows }) => {
-        if(rows.length===0){
-            return Promise.reject({ status: 404, msg: "not found" });
-        }
-      });
-    }
\ No newline at end of file
+  const queryStr = format(`SELECT * FROM articles;`);
+  return db.query(queryStr).then(({ rows }) => {
+    if (rows.length === 0) {
+      return Promise.reject({ status: 404, msg: "not found" });
+    }
+  });
+};
+
+exports.addCommentById = (newComment, article_id) => {
+  const { username, body } = newComment;
+  const regex = /\D/;
+  const isNonDigit = regex.test(article_id);
+  if (article_id <= 0 || isNonDigit) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
```

Comment:
You have added this functionality, but there's no test for it 🔍 
It would save a lot of code throughout your project if you passed through the wrong datatype to the query and caught the resulting PSQL error in your error handling middleware, rather than trying to catch it beforehand.